arrow/arrow_test.go

```diff
@@ -0,0 +1,130 @@
+package arrow_test
+
+import (
+	"encoding/hex"
+	"log"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tarantool/go-tarantool/v2"
+	"github.com/tarantool/go-tarantool/v2/arrow"
+	"github.com/tarantool/go-tarantool/v2/test_helpers"
+)
+
+var isArrowSupported = false
+
+var server = "127.0.0.1:3013"
+var dialer = tarantool.NetDialer{
+	Address:  server,
+	User:     "test",
+	Password: "test",
+}
+var space = "testArrow"
+
+var opts = tarantool.Opts{
+	Timeout: 5 * time.Second,
+}
+
+func skipIfArrowUnsupported(t *testing.T) {
+	t.Helper()
+	if !isArrowSupported {
+		t.Skip("Skipping test for Tarantool without Arrow support in msgpack")
+	}
+}
+
+// TestInsert uses Arrow sequence from Tarantool's test .
+// nolint:lll
+// See: https://github.com/tarantool/tarantool/blob/master/test/box-luatest/gh_10508_iproto_insert_arrow_test.lua
+func TestInsert_invalid(t *testing.T) {
+	skipIfArrowUnsupported(t)
+
+	arrows := []struct {
+		arrow    string
+		expected string
+	}{
+		{
+			"",
```

Comment:
Is it valid value from a Tarantool point of view? Why do we validate this value but not the others?

Reply:
This test to catch the error from `MakeArrow()` method for case of empty data.
What other values you are offering to check?

Reply:
I’ve offered do not check some cases, but check another. Let’s server validate values in this case too.

Reply:
On the other case we could save error in return values even it is never returned. It allows us to add some validation in the future without breaking of backward compatibility.

Reply:
Removed length validation in `MakeArrow`